src/wallet/rpcdump.cpp

```diff
@@ -324,6 +324,98 @@ RPCHelpMan importaddress()
     };
 }
 
+static void RemoveAddress(CWallet*, const CTxDestination&, bool);
+static void RemoveScript(CWallet* const pwallet, const CScript& script, bool is_redeem_script, bool purge_txns) EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet)
+{
+    isminetype ismine = pwallet->IsMine(script);
+    if (!is_redeem_script && ismine == ISMINE_SPENDABLE) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "The wallet contains the private key for this address");
+    } else if (!is_redeem_script && ismine == ISMINE_NO) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "The wallet does not contain this address or script");
+    }
+    CHECK_NONFATAL(ismine == ISMINE_WATCH_ONLY || is_redeem_script);
+
+    LegacyScriptPubKeyMan& spk_man = EnsureLegacyScriptPubKeyMan(*pwallet);
+
+    if (spk_man.HaveWatchOnly(script) && !spk_man.PurgeWatchOnly(script)) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "Error removing address/script from wallet");
+    }
+    if (is_redeem_script) {
+        CScriptID scriptID = CScriptID(script);
+        if (!spk_man.HaveCScript(scriptID)) {
+            throw JSONRPCError(RPC_WALLET_ERROR, "Error removing p2sh redeemScript from wallet");
```

Comment:
In commit "rpc: Added ability to remove watch only addresses" (b8eb5880693358951229512987beffcff51e0ecf)

Important: It seems dangerous to do this error check and raise an error after calling PurgeWatchOnly above. It would be better to check for this error condition early so the remove either fully fails or fully succeeds instead of half removing the redeem script.